### ISSUE TITLE
not working if container has display: flex

### ISSUE THREAD

**Issue by tadeusgregorian** [jasonslyvia/react-lazyload#96]
if the container element has the css style display:flex,
lazyloading is not working at all.

**Comment by jasonslyvia** [comment]
Interesting point, by not working you mean the `overflow` mode of LazyLoad not working, right?

**Comment by tadeusgregorian** [comment]
Yes, exactly. Overflow mode is not working with a flexbox-container. Took me a while to find the reason. I changed to display: block .

**Comment by jasonslyvia** [comment]
Thanks for reporting this, I think this should be fixed somehow.

**Comment by Redmega** [comment]
Would love to use this instead of trying to implement react-virtualized but I'm facing this issue right now! Any idea on the cause or possible fixes?

**Comment by jasonslyvia** [comment]
@Redmega I believe it's becasue of how we wrongly detect scrollParent for flex container, see https://github.com/jasonslyvia/react-lazyload/blob/0fb5a4552860a4c909daf40f22a64924cd1a039d/src/utils/scrollParent.js

**Comment by Nosherwan** [comment]
I am having the same issue, would love to have this fixed.

**Comment by larrydahooster** [comment]
Facing the same issue. Anyhow I wanted to switch over from react-lazy-load to react-lazyload because of the foreUpdate support. But they seem to have a similar logic for scrollParent detection which does not have trouble with flex: https://github.com/loktar00/react-lazy-load/blob/master/src/utils/parentScroll.js

**Comment by Hongbo-Miao** [comment]
I have same issue when using flex. Right now I am using `forceCheck();` as a walkaround way

**Comment by AncientPixel** [comment]
Just wanted to chip in here to show my support for the fixing of this issue, I have a flex based simple card view and having the same problem as stated by multiple people above.
Thanks for the awesome package.

**Comment by jfgen** [comment]
Yeah I'm also getting this issue, and it also happens when the items I'm loading have float rules. I believe the reasoning behind it is the same.

**Comment by BengBu-YueZhang** [comment]
同样的情况

**Comment by rxl881** [comment]
I'm unsure if this will fix all of the issues being discussed here, but I think that @jasonslyvia is correct (in the earlier comment) that the scrollParent detection is broken and that the trivial update in https://github.com/jasonslyvia/react-lazyload/pull/167 might fix that.

**Comment by jamsterdam29** [comment]
@Hongbo-Miao can you explain how you used ```forceCheck();``` as a solution?

I also would like to use this with a flex container but assume I am having the same issues.

**Comment by Hongbo-Miao** [comment]
Hi @jamsterdam29 I found out sometimes when scroll, it does not trigger correctly, it will skip some items, and those items remain unseen. I might do wrong. Since I have other higher priority tasks to finish, so I just stop using react-lazyload until I got time to try again.

**Comment by jaisonjjames** [comment]
Facing the same issue...

**Comment by gianpaj** [comment]
`forceCheck()` worked for me.

example when switching between Bootstrap (`reactstrap`) tabs:
```jsx
toggle = (tab: number) => {
  if (this.state.activeTab !== tab) {
    this.setState({ activeTab: tab }, () => {
      // hack to show tabs because react-lazyload is not activated for non-active tabs
      forceCheck();
    });
  }
}
```

**Comment by mcoevert** [comment]
Make sure you set the `scrollContainer` to a 'querySelectorable' string.
